examples/pytorch/grace/aug.py

```diff
@@ -5,29 +5,37 @@
 import dgl
 
 def aug(graph, x, feat_drop_rate, edge_mask_rate):
-    ng = drop_edge(graph, edge_mask_rate)
-    feat = drop_feat(x, feat_drop_rate)
-    ng = ng.add_self_loop()
+    n_node = graph.number_of_nodes()
 
-    return ng, feat
+    edge_mask = mask_edge(graph, edge_mask_rate)
+    feat = drop_feature(x, feat_drop_rate)
 
-def drop_edge(graph, drop_prob):
-    E = graph.num_edges()
+    ng = dgl.graph([])
+    ng.add_nodes(n_node)
+    src = graph.edges()[0]
+    dst = graph.edges()[1]
 
-    mask_rates = th.FloatTensor(np.ones(E) * drop_prob)
-    masks = th.bernoulli(1 - mask_rates)
-    edge_idx = masks.nonzero().squeeze(1)
+    nsrc = src[edge_mask]
+    ndst = dst[edge_mask]
+    ng.add_edges(nsrc, ndst)
 
-    sg = dgl.edge_subgraph(graph, edge_idx, relabel_nodes=False)
-    
-    return sg
+    ng = ng.add_self_loop()
 
-def drop_feat(x, drop_prob):
-    D = x.shape[1]
-    mask_rates = th.FloatTensor(np.ones(D) * drop_prob)
-    masks = th.bernoulli(1 - mask_rates)
+    return ng, feat
 
+def drop_feature(x, drop_prob):
+    drop_mask = th.empty((x.size(1),),
+                        dtype=th.float32,
+                        device=x.device).uniform_(0, 1) < drop_prob
     x = x.clone()
```

Comment:
Why do we need `x.clone()`?

Reply:
Otherwise the input feature mask would be changed.

Reply:
Ok